### PR TITLE
Access log valve

### PR DESCRIPTION
**Pull request by jfdenise**
This is a trio of three pull requests. These are:

* https://github.com/wildfly/wildfly-cekit-modules/pull/48
* https://github.com/wildfly/temp-eap-modules/pull/34
* https://github.com/wildfly/temp-openshift-image/pull/21

### CODE REVIEW

**Comment by jfdenise** [comment]
@yersan , that is @kabir PR with fixes for your review,

File: jboss/container/wildfly/launch/access-log-valve/added/launch/access_log_valve.sh [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+      testXpathExpression "${xpath}" "serverNamesRet" "serverNames"
+      if [ "${serverNamesRet}" -ne 0 ]; then
+        echo "You have set ENABLE_ACCESS_LOG=true to add the access-log valve. Fix your configuration to contain at least one server in the undertow subsystem for this to happen." >> ${CLI_SCRIPT_ERROR_FILE}
+        return
+      fi
+
+      # Not having any server hosts is an error
+      local hostsRet
+      local xpath="\"//*[local-name()='subsystem' and starts-with(namespace-uri(), 'urn:jboss:domain:undertow:')]/*[local-name()='server']/*[local-name()='host']\""
+      testXpathExpression "${xpath}" "ret"
```
Comment (yersan) [inline jboss/container/wildfly/launch/access-log-valve/added/launch/access_log_valve.sh]: This should be `testXpathExpression "${xpath}" "hostsRet"`

**Comment by kabir** [comment]
@jfdenise Feel free to merge this if the behave tests passed. It would be good if you fix @yersan 's comment, but there is no need to build the image and run the tests again after that since that code path is untestable (or can only be tested with great difficulty which I don't want to do now) as mentioned in https://github.com/wildfly/temp-eap-modules/pull/34/files#diff-4d9336efd10f04e153e582c48c7007dcR16

**Comment by jfdenise** [comment]
@kabir , tests pass, fixed last comment.

**Pull request merged**